src/crafting.h

```diff
@@ -69,6 +69,8 @@ struct recipe {
         bool hidden;
     };
     std::vector<bookdata_t> booksets;
+    std::set<std::string> flags;
```

Comment:
Adding two bools or a float makes this too single-purpose. I anticipate adding more flags to recipes in the future. An array of strings is how flags are handled in every other object type we deal with. @kevingranade said he was "leaning toward a flag".